sdk/identity/azure-identity/src/main/java/com/azure/identity/AzurePipelinesCredentialBuilder.java

```diff
@@ -64,26 +64,25 @@ public AzurePipelinesCredentialBuilder systemAccessToken(String systemAccessToke
      * @return an instance of the {@link AzurePipelinesCredential}.
      */
     public AzurePipelinesCredential build() {
-        ValidationUtil.validate(getClass().getSimpleName(),
-            LOGGER,
-            Arrays.asList("clientId", "tenantId", "serviceConnectionId", "systemAccessToken"),
-            Arrays.asList(this.clientId, this.tenantId, this.serviceConnectionId, this.systemAccessToken));
-
         Configuration configuration = identityClientOptions.getConfiguration();
         if (configuration == null) {
             configuration = Configuration.getGlobalConfiguration();
         }
-        String teamFoundationCollectionUri = configuration.get("SYSTEM_TEAMFOUNDATIONCOLLECTIONURI");
-        String teamProjectId = configuration.get("SYSTEM_TEAMPROJECTID");
-        String planId = configuration.get("SYSTEM_PLANID");
-        String jobId = configuration.get("SYSTEM_JOBID");
-
-        ValidationUtil.validate(getClass().getSimpleName(), LOGGER,
-            Arrays.asList("SYSTEM_TEAMFOUNDATIONCOLLECTIONURI", "SYSTEM_TEAMPROJECTID", "SYSTEM_PLANID", "SYSTEM_JOBID"),
-            Arrays.asList(teamFoundationCollectionUri, teamProjectId, planId, jobId));
 
-        String requestUrl = String.format("%s%s/_apis/distributedtask/hubs/build/plans/%s/jobs/%s/oidctoken?api-version=%s&serviceConnectionId=%s",
-            teamFoundationCollectionUri, teamProjectId, planId, jobId, OIDC_API_VERSION, serviceConnectionId);
-        return new AzurePipelinesCredential(this.clientId, this.tenantId, requestUrl, systemAccessToken, identityClientOptions.clone());
+        String oidcEndpoint = configuration.get("SYSTEM_OIDCREQUESTURI");
+        try {
+            ValidationUtil.validate(getClass().getSimpleName(),
+                LOGGER,
+                Arrays.asList("clientId", "tenantId", "serviceConnectionId", "systemAccessToken", "oidcEndpoint"),
+                Arrays.asList(clientId, tenantId, serviceConnectionId, this.systemAccessToken, oidcEndpoint));
+        } catch (IllegalArgumentException e) {
+            throw LOGGER.logExceptionAsError(new CredentialUnavailableException(
```

Comment:
We don't throw Cred Unavailable exception when constructing the credential.
Validating and throwing on user provided input params is fine in the build method.
What logic are other languages following here ?